src/lib/x509/x509_ext.h

```diff
@@ -500,6 +500,65 @@ class OCSP_NoCheck final : public Certificate_Extension {
       void decode_inner(const std::vector<uint8_t>&) override;
 };
 
+/**
+* TNAuthList extension
+*
+* RFC8226 Secure Telephone Identity Credentials
+*   https://www.rfc-editor.org/rfc/rfc8226#section-9
+*/
+class BOTAN_PUBLIC_API(3, 5) TNAuthList final : public Certificate_Extension {
+   public:
+      class Entry final : public ASN1_Object {
+         public:
+            enum Type { TN_ServiceProviderCode = 0, TN_TelephoneNumberRange, TN_TelephoneNumber };
```

Comment:
Currently the code implicitly assumes that the values of this `enum` match the ASN.1. Let's make this a bit more explicit by directly initializing each with their respective value, along with a comment that these values match the respective tagging of `TNEntry` in https://datatracker.ietf.org/doc/html/rfc8226#section-9